tests/lib/linter/linter.js

```diff
@@ -6737,6 +6760,49 @@ var a = "test2";
         });
     });
 
+    describe("options", () => {
+        it("rules should apply meta.defaultOptions and ignore schema defaults", () => {
+            linter.defineRule("my-rule", {
+                meta: {
+                    defaultOptions: [{
+                        inBoth: "from-default-options",
+                        inDefaultOptions: "from-default-options"
+                    }],
+                    schema: {
+                        type: "object",
+                        properties: {
+                            inBoth: { default: "from-schema", type: "string" },
+                            inDefaultOptions: { type: "string" },
+                            inSchema: { default: "from-schema", type: "string" }
+                        },
+                        additionalProperties: false
+                    }
+                },
+                create(context) {
+                    return {
+                        Program(node) {
+                            context.report({ node, message: JSON.stringify(context.options[0]) });
+                        }
+                    };
+                }
+            });
+
+            const config = {
+                rules: {
+                    "my-rule": "error"
+                }
+            };
+
+            const code = "";
+            const messages = linter.verify(code, config);
+
+            assert.deepStrictEqual(
+                JSON.parse(messages[0].message),
+                { inBoth: "from-default-options", inDefaultOptions: "from-default-options" }
```

Comment:
Spinning out from https://github.com/eslint/eslint/pull/17656#discussion_r1475266210:

> Should we mention how `defaultOptions` interacts with `default` values from the schema and which one is recommended?

I'd like to add a unit test that shows the end-to-end behavior. But I can't figure out where that unit test should go. Does anybody have any advice?

This test here shows the same behavior as on `main`: for custom rules, the `default`s in `schema` properties aren't applied.

Reply:
Are you asking if it belongs in `linter.js`? Or are you asking where it belongs within `linter.js`?

It would definitely go in `linter.js`, and I'd just put a new `describe` for this.

Note that `linter.js` is split into two sections: eslintrc config and flat config, so you need to add a test in both sections.

Reply:
The question here on boundaries of when to apply schema defaults is relevant, I think: https://github.com/eslint/eslint/pull/17656/files#r1491809729